build-system-abstraction.rst

```diff
@@ -0,0 +1,360 @@
+
+schema
+    The version of the schema. This PEP defines version 1.
+    Defaults to 1
+bootstrap-requires
+    The requirements required to run the build tool itself.
+    
+    This should be a list of specifiers, in plain text form.
+    
+    e.g.::
+    
+        bootstrap-requires:
+          - requirement
+          - requirement[extra]
+          - requirement:python_version<"3"
+          - requirement[extra]:python_version>"2"
+          - requirement[extra]>2.5:python_version=="2.7"
+
+build-tool
+    A command to run to query the build tool for its complete interface.
+    The build tool should output on stdout a build tool description JSON document.
+    Stdin may not be read from, and stderr may be handled however the calling process
+    desires.
+
+build tool description
+----------------------
+
+The build tool description schema. Extra keys are ignored.
+
+schema
+    The version of the schema. This PEP defines version 1.
+    Defaults to 1
+build-requires
+    Command to run to query build requirements. Build requirements are
+    returned as a partially filled out PEP-426 [#pep426]_ JSON metadata for the project.
+    Additional data *may* be included, but the ``build_requires`` and ``metadata_version``
+    keys must be present.
+dist-info
+    Command to run to generate PEP-426 JSON metadata for the project.
+    The metadata should be output on stdout, stdin may not be consumed,
+    and stderr handling is up to the discretion of the calling process.
+    The build-requires for the project will be present in the Python
+    environment when the dist-info command is run.
+    pip would run dist-info just once to determine what other packages need to
+    be downloaded and installed.
+wheel
+    Command to run to build a wheel of the project. OUTPUT_DIR will be
+    set and point to an existing directory where the wheel should be output.
+    Stdin may not be consumed, stdout and stderr handling is at the discretion
+    of the calling process. The build-requires for the project will be
+    present in the Python environment when the wheel command is run.
+    Only one file should be output - if more are output then pip would
+    pick an arbitrary one to consume.
+develop
+    Command to do an in-place 'development' installation of the project.
+    Stdin may not be consumed, stdout and stderr handling is at the discretion
+    of the calling process. The build-requires for the project will be
+    present in the Python environment when the develop command is run.
+    
+    Not all build systems will be able to perform develop installs. If a build
+    system cannot do develop installs, then this key can be omitted.
+    
+    Note tht when it is omitted, commands like ``pip install -e foo``
+    will be unable to complete.
+provided-by
+    The distribution name that define this build system interface.
+    This is used to facilitate caching the build tool output to avoid requiring
```

Comment:
I had to read this three times. ``build tool output`` here means the output of running the entry under ``build-tool`` from ``pypa.yaml``. (it could also have meant a wheel/sdist/... - a build tool outputs a lot of things)